docs/source/user-guide/dataframe.md

```diff
@@ -0,0 +1,278 @@
+| collect                    | Executes this DataFrame and collects all results into a vector of RecordBatch.                                              |
+| collect_partitioned        | Executes this DataFrame and collects all results into a vector of vector of RecordBatch maintaining the input partitioning. |
+| execute_stream             | Executes this DataFrame and returns a stream over a single partition.                                                       |
+| execute_stream_partitioned | Executes this DataFrame and returns one stream per partition.                                                               |
+| show                       | Execute this DataFrame and print the results to stdout.                                                                     |
+| show_limit                 | Execute this DataFrame and print a subset of results to stdout.                                                             |
+| write_csv                  | Execute this DataFrame and write the results to disk in CSV format.                                                         |
+| write_json                 | Execute this DataFrame and write the results to disk in JSON format.                                                        |
+| write_parquet              | Execute this DataFrame and write the results to disk in Parquet format.                                                     |
+
+## Other DataFrame Methods
+
+| Function        | Notes                                                                                                                                                        |
+| --------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| explain         | Return a DataFrame with the explanation of its plan so far.                                                                                                  |
+| registry        | Return a `FunctionRegistry` used to plan udf's calls.                                                                                                        |
+| schema          | Returns the schema describing the output of this DataFrame in terms of columns returned, where each column has a name, data type, and nullability attribute. |
+| to_logical_plan | Return the logical plan represented by this DataFrame.                                                                                                       |
+
+# Expressions
+
+DataFrame methods such as `select` and `filter` accept one or more logical expressions and there are many functions
+available for creating logical expressions. These are documented below.
+
+Expressions can be chained together using a fluent-style API:
+
+```rust
+col("a").gt(lit(5)).and(col("b").lt(lit(7)))
+```
+
+## Identifiers
```

Comment:
Since these are not data frame specific, maybe we should put them into a different guide -- and reference that here

Something like `docs/source/user-guide/expressions.md` perhaps

This is just a suggestion, I think it would be fine to do in a follow on PR or never

Reply:
I will move into a separate page as a follow-on. How would you see users using expressions outside the context of DataFrame usage?

Reply:
We use them all the time in IOx when we need to make `Exprs` -- for example to create `LogicalPlans` / use the `LogicalPlanBuilder`. I also think they are useful for writing tests when people are writing extensions for DataFusion